ironic/resource_ironic_deployment.go

```diff
@@ -125,7 +134,7 @@ func resourceDeploymentCreate(d *schema.ResourceData, meta interface{}) error {
 }
 
 // fetchFullIgnition gets full igntion from the URL and cert passed to it and returns userdata as a string
-func fetchFullIgnition(userDataURL string, userDataCaCert string) string {
+func fetchFullIgnition(userDataURL string, userDataCaCert string, userDataHeaders map[string]interface{}) (string, error) {
```

Comment:
Test needs updating: 

```
ironic/resource_ironic_deployment_test.go:167:12: assignment mismatch: 1 variable but fetchFullIgnition returns 2 values

ironic/resource_ironic_deployment_test.go:167:32: not enough arguments in call to fetchFullIgnition
```

Reply:
Ah thanks, I couldn't figure out how to get the tests to run locally - it'd be nice if we could somehow decouple the unit tests from the integration tests which require a local ironic etc